protocols/tests/test_migration/test_migration_reports_4_0_0_to_reports_3_0_0.py

```diff
@@ -45,3 +45,84 @@ def test_migrate_rd_clinical_report(self, fill_nullables=True):
 
     def test_migrate_rd_clinical_report_fill_nullables_false(self):
         self.test_migrate_rd_clinical_report(fill_nullables=False)
+
+    def test_migrate_interpretation_request_rd(self):
+        ir_rd_4 = self.get_valid_object(
+            object_type=self.old_model.InterpretationRequestRD,
+            version=self.version_4_0_0,
+            fill_nullables=True,
+        )
+        ir_rd_3 = MigrateReports400To300().migrate_interpretation_request_rd(old_instance=ir_rd_4)
+        self.assertIsInstance(ir_rd_3, self.new_model.InterpretationRequestRD)
+        self.assertTrue(ir_rd_3.validate(ir_rd_3.toJsonDict()))
+
+    def test_migrate_pedigree(self):
+        pedigree_v4 = self.get_valid_object(
+            object_type=self.old_model.Pedigree,
+            version=self.version_4_0_0,
+            fill_nullables=True
+        )
+        pedigree_v3 = MigrationParticipants100ToReports().migrate_pedigree(old_pedigree=pedigree_v4)
+        self.assertIsInstance(pedigree_v3, self.new_model.Pedigree)
+        self.assertTrue(pedigree_v3.validate(pedigree_v3.toJsonDict()))
+
+        for a, d in zip(pedigree_v3.analysisPanels, pedigree_v3.diseasePenetrances):
+            self.assertIsInstance(a, self.new_model.AnalysisPanel)
+            self.assertIsInstance(d, self.new_model.DiseasePenetrance)
+
+    def test_migrate_analysis_panel(self):
+        panel_v4 = self.get_valid_object(
+            object_type=self.old_model.AnalysisPanel,
+            version=self.version_4_0_0,
+            fill_nullables=True,
+        )
+        self.assertIsInstance(panel_v4, self.old_model.AnalysisPanel)
+        self.assertTrue(panel_v4.validate(panel_v4.toJsonDict()))
+        panel_v3 = MigrationParticipants100ToReports().migrate_analysis_panel(old_panel=panel_v4)
+        self.assertIsInstance(panel_v3, self.new_model.AnalysisPanel)
+        self.assertTrue(panel_v3.validate(panel_v3.toJsonDict()))
+
+    def test_migrate_file(self):
+        file_v4 = self.get_valid_object(
+            object_type=self.old_model.File,
+            version=self.version_4_0_0,
+            fill_nullables=True,
+        )
+        self.assertIsInstance(file_v4, self.old_model.File)
+        self.assertTrue(file_v4.validate(file_v4.toJsonDict()))
+        file_v3 = MigrateReports400To300().migrate_file(old_file=file_v4)
+        self.assertIsInstance(file_v3, self.new_model.File)
+        self.assertTrue(file_v3.validate(file_v3.toJsonDict()))
+
+    def test_migrate_member_to_participant(self):
+        # Can not run with fill_nullables False as the new model has non-nullable fields
+        # when the old model has nullable fields (for the same fields)
```

Comment:
Our objective is make any possible payload not to fail these migrations. When this situation described happens, we set default values or if it is really impossible we need to control that error and show an appropriate message, but the later whould be extreme cases. So, testing this is a must.

I will do it.